R/evidence_base_panel.R

```diff
@@ -0,0 +1,103 @@
+#' Module UI for 'evidence base' panel.
+#' 
+#' @param id ID of the module
+#' @return Div for the home page
+evidence_base_panel_ui <- function(id) {
+  ns <- NS(id)
+  div(
+    column(
+      width = 5,
+      align = 'center',
+      bsCollapse(
+        id = ns("EvidenceBase"),
+        open = "Current Evidence Base",
+        bsCollapsePanel(
+          title = "Current Evidence Base",
+          style = 'primary',
+          h6("This panel presents the current evidence base from which the sample size calculations are based on. If you wish to change this, please alter the synthesis options above accordingly."),
+          withSpinner(
+            type = 6,
+            plotOutput(outputId = ns("EvBase"))
+          ),
+          radioButtons(
+            inputId = ns("EvBase_choice"),
+            label = NULL,
+            choices = c(
+              "Frequentist MA" = "freq",
+              "Bayesian MA" = "bayes"
+            ),
+            inline = TRUE
+          ),
+          fluidRow(
+            div(
+              style = "display: inline-block;",
+              downloadButton(outputId = ns('evbase_download'), label = "Download forest plot")
+            ),
+            div(
+              style = "display:inline-block; width: 10px;",
+              br()
+            ),
+            div(
+              style = "display: inline-block;",
+              radioButtons(inputId = ns('evbase_choice'), label = NULL, choices = c('pdf', 'png'), inline = TRUE)
+            )
+          )
+        )
+      )
+    )
+  )
+}
+
+#' Module server for 'evidence base' panel.
+#' 
+#' @param id ID of the module
+#' @param freqpair Reactive object of frequentist analysis
+#' @return Input for choice of evidence base
+evidence_base_panel_server <- function(id, freqpair) {     
+  moduleServer(id, function(input, output, session) {
+    
+    # Forest plot of current evidence base #
+    output$EvBase <- renderPlot({
+      if (input$EvBase_choice == 'bayes') {
+        NoBayesian()
+      } else if (input$EvBase_choice == 'freq') {
+        if (freqpair()$MA$MA.Fixed$measure %in% c('OR', 'RR')) {
+          forest.rma(freqpair()$MA$MA.Fixed, atransf = exp, ylim = -2.5)
+          addpoly(freqpair()$MA$MA.Random)
+        } else {
+          forest.rma(freqpair()$MA$MA.Fixed, ylim = -2.5)
+          addpoly(freqpair()$MA$MA.Random)
+        }
+        title("Forest plot of studies and overall pooled estimates")
+      }
+    })
+  
+    # Download #
+    output$evbase_download <- downloadHandler(
+      filename = function() {
+        paste0("EvidenceBase.", input$evbase_choice)
+      },
+      content = function(file) {
+        if (input$evbase_choice == 'pdf') {
+          pdf(file = file)
+        } else {
+          png(file = file)
+        }
+        if (input$EvBase_choice == 'freq') {
```

Comment:
What is meant to happen if `EvBase_choice` is not `'freq'`?

Reply:
This is legacy from when I was hoping to get Bayesian and frequentist running from 'step 2' onwards. Currently, the app doesn't show the Bayesian evidence base and shows a warning that the feature is not ready - so there is nothing to download in that case.

So, I guess until I add Bayesian, this if statement is not of any use - so I've removed it - let me know if this is not ideal?

Reply:
As discussed, this still can lead to an erroneous download. I think the choices are:
- 1 - Disable the "Bayesian MA" radio button, with a tooltip to explain that it isn't available yet.
- 2 - Rename the download button to "Download frequentist forest plot".
  - Ideally with the button disabled until the frequentist forest plot is available.
- 3 - Hide the forest plot and download button until the frequentist evidence base is available.
  - Or replace the contents with "Only available for frequentist analysis.
  - The button would probably need renaming in this case too.

1 is probably the most straightforward while retaining the original intent. It might need to special CSS to get the individual radio button disabled, but I've done it with a checkbox group, so it shouldn't be too hard.